flaml/autogen/agent/math_user_proxy_agent.py

```diff
@@ -0,0 +1,410 @@
+3. Wait for me to give the results.
+4. Continue if you think the result is correct. If the result is invalid or unexpected, please correct your query or reasoning.
+
+After all the queries are run and you get the answer, put the final answer in \\boxed{}.
+
+Problem: """,
+    # use python step by step
+    "python": """Let's use Python to solve a math problem.
+
+Query requirements:
+You should always use the 'print' function for the output and use fractions/radical forms instead of decimals.
+You can use packages like sympy to help you.
+You must follow the formats below to write your code:
+```python
+# your code
+```
+
+Please follow this process:
+1. Solve the problem step by step (do not over-divide the steps).
+2. Take out any queries that can be asked through Python (for example, any calculations or equations that can be calculated).
+3. Wait for me to give the results.
+4. Continue if you think the result is correct. If the result is invalid or unexpected, please correct your query or reasoning.
+
+After all the queries are run and you get the answer, put the answer in \\boxed{}.
+
+Problem: """,
+}
+
+
+class MathUserProxyAgent(UserProxyAgent):
+    """(Experimental) A MathChat agent that can handle math problems."""
+
+    MAX_CONSECUTIVE_AUTO_REPLY = 15  # maximum number of consecutive auto replies (subject to future change)
+
+    def __init__(
+        self,
+        name="MathChatAgent",  # default set to MathChatAgent
+        system_message="",
+        work_dir=None,
+        human_input_mode="NEVER",  # Fully automated
+        max_consecutive_auto_reply=None,
+        is_termination_msg=None,
+        use_docker=True,
+        max_invalid_q_per_step=3,  # a parameter needed in MathChat
+        **config,
+    ):
+        """
+        Args:
+            name (str): name of the agent
+            system_message (str): system message to be sent to the agent
+            work_dir (str): working directory for the agent
+            human_input_mode (str): whether to ask for human inputs every time a message is received.
+                Possible values are "ALWAYS", "TERMINATE", "NEVER".
+                (1) When "ALWAYS", the agent prompts for human input every time a message is received.
+                    Under this mode, the conversation stops when the human input is "exit",
+                    or when is_termination_msg is True and there is no human input.
+                (2) When "TERMINATE", the agent only prompts for human input only when a termination message is received or
+                    the number of auto reply reaches the max_consecutive_auto_reply.
+                (3) When "NEVER", the agent will never prompt for human input. Under this mode, the conversation stops
+                    when the number of auto reply reaches the max_consecutive_auto_reply or or when is_termination_msg is True.
```

Comment:
I will try implementing an executor based on ipykernel. I would like to keep it in the core library.

Reply:
ipykernel execution should be considered as a feature in UserProxyAgent because it's generic. Then the implementation of MathUserProxyAgent can be simplified.

Reply:
I am looking into it.